src/cast_core/src/CrudActor.cpp

```diff
@@ -699,6 +699,65 @@ struct FindOneOperation : public BaseOperation {
     metrics::Operation _operation;
 };
 
+struct AggregateOperation : public BaseOperation {
+    struct Pipeline {
+        Pipeline(const Node& node, PhaseContext& context, ActorId id) {
+            if (!node.isSequence()) {
+                BOOST_THROW_EXCEPTION(InvalidConfigurationException("'Pipeline' must be an array"));
+            }
+            for (auto&& [_, stageNode] : node) {
+                DocumentGenerator stage = stageNode.to<DocumentGenerator>(context, id);
+                pipeline.append_stage(stage());
+            }
+        }
+
+        bsoncxx::document::value copyPipelineToDocument() const {
+            auto arrayView = pipeline.view_array();
+            bsoncxx::document::view docView{arrayView.data(), arrayView.length()};
+            return bsoncxx::document::value{docView};
+        }
+
+        mongocxx::pipeline pipeline;
+    };
+
+    AggregateOperation(const Node& opNode,
+                       bool onSession,
+                       mongocxx::collection collection,
+                       metrics::Operation operation,
+                       PhaseContext& context,
+                       ActorId id)
+        : BaseOperation(context, opNode),
+          _onSession{onSession},
+          _collection{std::move(collection)},
+          _operation{operation},
+          _pipeline{opNode["Pipeline"].to<Pipeline>(context, id)} {
```

Comment:
Nice!

Reply:
😄 

I thought the API for converting the YAML to typed constructs is pretty nifty.